### ISSUE TITLE
REST API: Get all chapters of a media resource

### ISSUE THREAD

**Issue by jthomsen** [linkedtv/platform#27]
GET /mediaresource/{uuid}/chapter

**Comment by jthomsen** [comment]
SPARQL Query:

``` sparql
PREFIX linkedtv: <http://data.linkedtv.eu/ontologies/core#>
PREFIX ma: <http://www.w3.org/ns/ma-ont#>
PREFIX nsa: <http://multimedialab.elis.ugent.be/organon/ontologies/ninsuna#>
PREFIX oa: <http://www.w3.org/ns/oa#>
SELECT DISTINCT ?item
WHERE {
 ?mediafragment ma:isFragmentOf ?mediaresource . 
 ?mediafragment a ma:MediaFragment . 
 ?annotation a oa:Annotation .  
 ?annotation oa:hasBody ?item . 
 ?annotation oa:hasTarget ?mediafragment . 
 ?item a linkedtv:Chapter .  
}
```

**Comment by jthomsen** [comment]
has been realized with the above SPARQL query; retrieves chapters from uncurated annotations; sample call:

http://data.linkedtv.eu/mediaresource/adb65e0a-642b-432f-aa86-c296dab0375a/chapter

**Comment by rtroncy** [comment]
What is the purpose of the last triple pattern (?mediafragment nsa:temporalStart ?start ) ?

What is the relationship between ?chapter and ?item ? It seems to me  you just need one variable

**Comment by jthomsen** [comment]
yes, you're right - thanks for checking! this was indeed a copy / paste error; changed it

**Comment by jblom** [comment]
Hi Jan,

thanks for this. I will check later myself but: do the results also include the provenance information? It's important for the ET and the player to check whether these are automated or curated chapters.

(this basically applies for all the other API calls)

Best,

Jaap

**Comment by jthomsen** [comment]
Hi Jaap, 
well at the moment they just return a list of chapter ids from the standard graph, i.e. the non-curated ones; think I could add something like /mediaresource/{uuid}/chapter/curated for the curated ones
would that be ok?

Best
Jan

**Comment by rtroncy** [comment]
There should not be a "standard graph" whatever it means, since all annotations have PROV statements. Creating a different URI pattern for all possible PROV values does not seem a good idea. Why does /mediaresource/{uuid}/chapter/ simply not return ALL chapters with PROV statements telling how the chapter has been created?

**Comment by jblom** [comment]
Hi Raphael,

this is due to the fact that the ET still saves in a different graph. I will need to update the saving code of the ET to integrate the data in the 'main' LinkedTV graph (still other things were more important up till now). Hmm I guess it makes sense, if I will do this first. In any case, it's necessary to return the prov data for all of the API calls (if applicable).
